cmd/embedded-cluster/install.go

```diff
@@ -551,31 +554,42 @@ func runOutro(c *cli.Context, applier *addons.Applier, cfg *k0sconfig.ClusterCon
 func maybeAskAdminConsolePassword(c *cli.Context) (string, error) {
 	defaultPassword := "password"
 	userProvidedPassword := c.String("admin-console-password")
-	if c.Bool("no-prompt") {
-		if userProvidedPassword != "" {
-			return userProvidedPassword, nil
-		} else {
-			logrus.Infof("The Admin Console password is set to %s", defaultPassword)
-			return defaultPassword, nil
-		}
-	}
+	// If there's a user provided password we'll try that first
 	if userProvidedPassword != "" {
+		if !validateAdminConsolePassword(userProvidedPassword, "", false) {
+			return "", fmt.Errorf("unable to set the Admin Console password")
+		}
 		return userProvidedPassword, nil
 	}
+	// No user provided password but prompt is disabled so we set our default password
+	if c.Bool("no-prompt") {
+		logrus.Infof("The Admin Console password is set to %s", defaultPassword)
+		return defaultPassword, nil
+	}
 	maxTries := 3
 	for i := 0; i < maxTries; i++ {
-		promptA := prompts.New().Password("Set the Admin Console password:")
+		promptA := prompts.New().Password(fmt.Sprintf("Set the Admin Console password (minimum %d characters):", minAdminPasswordLength))
 		promptB := prompts.New().Password("Confirm the Admin Console password:")
 
-		if promptA == promptB {
-			// TODO: Should we add extra password validation here? e.g length, complexity etc
+		if validateAdminConsolePassword(promptA, promptB, true) {
 			return promptA, nil
 		}
-		logrus.Info("Passwords don't match. Please try again.")
 	}
 	return "", fmt.Errorf("unable to set the Admin Console password after %d tries", maxTries)
 }
 
+func validateAdminConsolePassword(password, passwordCheck string, mustMatch bool) bool {
```

Comment:
The only time this is called with `mustMatch: false` is when it's provided via the CLI. Would it make more sense for that function call to just supply the password twice to the function?

Reply:
Personally, I prefer being explicit about the functionality VS having it be dependent on behaviour which is non-explicit. But happy to change it if y'all think differently.

Reply:
my feeling is that 'mustMatch' isn't particularly explicit - if we're going to do something like that, we should have two functions one for 'passwordLength' and another for 'passwordsMatch' IMO

@sgalsaleh ?

Reply:
Went with your original suggestion via - https://github.com/replicatedhq/embedded-cluster/pull/1262/commits/9ab78499e3eefeed87cf8bb11998303c503ef779 - let me know what you think @laverya